addon/components/donate-banner/component.js

```diff
@@ -0,0 +1,111 @@
+import Ember from 'ember';
+import layout from './template';
+import AnalyticsMixin from 'ember-osf/mixins/analytics';
+
+/**
+ * @module ember-osf
+ * @submodule components
+ */
+
+/**
+ * Display the donate banner.  Has an automated script to determine
+ * what the banner should look like based on the the current date.
+ *
+ * Sample usage:
+ * ```handlebars
+ * {{donate-banner}}
+ * ```
+ *
+ * @class donate-banner
+ */
+export default Ember.Component.extend(AnalyticsMixin, {
+
+    init() {
+        this._super();
+
+        let currentDate = this.get('currentDate') ? this.get('currentDate') : new Date();
+
+        let changeDates = this.get('changeDates');
+
```

Comment:
Suggestion:
```
let endDate = changeDates.pop().date;
```
Then you just start at `changeDates.length - 1` below and compare to endDate.